www/src/pages/blog/astro-021-preview.astro

```diff
@@ -0,0 +1,119 @@
+
+        If you're curious to learn more, you can read more about this transition from my perspective as Snowpack maintainer [here.](https://dev.to/fredkschott/5-more-things-i-learned-building-snowpack-to-20-000-stars-5dc9)
+
+
+        ## Hello, WASM!
+
+        Astro 0.21 features another huge low-level improvement: the brand new [@astrojs/compiler](https://github.com/snowpackjs/astro-compiler-next). Astro's new compiler is written in Go and distributed as WASM. You can run it right in your browser, or on the server in Node.js and Deno. 
+
+        The new [@astrojs/compiler](https://github.com/snowpackjs/astro-compiler-next) unlocks:
+
+        - **Flexibility:** Run the compiler anywhere with WASM.
+        - **Speed:** Build sites faster with Go's compiled-language performance.
+        - **Stability:** Writing our own compiler allowed us to fix some long-standing bugs.
+
+        You can play with the new compiler today right in your browser at https://astro.build/play. This REPL is just one example of what is now possible when you have a fast, runs-anywhere compiler. 
+
+        Shout out to [Nate Moore](https://twitter.com/n_moore) who did a really incredible job with this project.
+
+
+        ## Components in Markdown
+
+        Our most requested feature ***by far*** has been the ability to use components directly in Markdown. After months of work, we're excited to announce that this feature is finally coming to Astro.
+
+        Starting in v0.21.0, you can import components inside of your Markdown frontmatter via an optional `setup` script. Once imported, your components can be used anywhere on the page:
+
+        <Code code={`---
+title: 'Astro 0.21 Early Preview'
+setup: |
+  import Logo from '../components/Logo.astro';
+  import ReactCounter from '../components/Counter.jsx';
+---
+
+# Astro now supports components in Markdown!
+
+<Logo />
+
+- Back to markdown here. 
+- Supports static Astro components.
+- Supports dynamic React/Vue/Svelte components!
+
+<ReactCounter start={0} client:load /> 
+`} lang="astro" />
+
+        This new `setup` script was designed for maximum flexibility. We'll keep improving this API going forward with planned support for default components, default layouts, and markdown component overrides.
+
+
+        ## HMR, meet HTML
+
+        Starting in v0.21.0, Astro will support full HMR for Astro components and pages. Change any `.astro` file in your codebase, and watch the dev server update the page without a full refresh and without losing any client state.
+
+        Astro has always supported powerful HMR updates for client-side JavaScript components like React, Preact, Svelte, Vue, and Solid.js. But adding this for Astro was a fun challenge because Astro components are just static HTML. Our "Zero JavaScript" approach meant that there was no "Astro runtime" to hook into for updates. We had to get creative.
```

Comment:
I'm curious whether this extends to markdown from Astro.fetchContent since today it seems like i need to restart the server to access updates 😅

Reply:
it should! I think that's a bug in the current `next` branch